flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BaseWriter.java

```diff
@@ -56,15 +57,15 @@
  * validated lazily.
  *
  * <p>Serializer's "init" method is called in the writer's constructor because the resulting {@link
- * Descriptor} is not serializable and cannot be propagated to machines hosting writer instances.
- * Hence, this derivation of descriptors must be performed during writer initialization.
+ * com.google.protobuf.Descriptors.Descriptor} is not serializable and cannot be propagated to
+ * machines hosting writer instances. Hence, this derivation of descriptors must be performed during
+ * writer initialization.
  *
  * @param <IN> Type of records to be written to BigQuery.
  */
 abstract class BaseWriter<IN> implements SinkWriter<IN> {
 
-    protected final Logger logger = LoggerFactory.getLogger(getClass());
-
+    protected final Logger logger = LoggerFactory.getLogger(BaseWriter.class);
```

Comment:
Why is this needed?
https://stackoverflow.com/questions/12160540/java-logging-with-abstract-classes

Reply:
Uniformity, should we change to `getClass()` for every class that uses a logger?

Reply:
`getClass()` should be used only when the class type is not known an compile time. BaseWriter in itself is an abstract class. The logs defined in it will be executed by its implementations (ALO and EXO writers) at runtime.